### PR TITLE
feat : Not opcode and updated readme

### PR DESCRIPTION
**Pull request by mahmudsudo**
1. #24 Not Opcode implemented with tests
2. readme updated to remove error when running command

### CODE REVIEW

File: src/codegen/operations.rs [hunk 1]
```diff
@@ -596,3 +598,47 @@ fn codegen_jumpdest<'c>(
 
     Ok((landing_block, landing_block))
 }
+
+// from the understanding of the not operator , A xor 1 == Not A
```
Comment (MegaRedHand) [inline src/codegen/operations.rs:602]: This is correct only for boolean values. Here we're working with 256-bit integers. For them, bitwise NOT is equivalent to `A xor 0xff...ff` (i.e. xor-ing against 256 set bits)

**Review by MegaRedHand — CHANGES_REQUESTED** [review]
Hey there, and thanks again for your contribution! I left you some comments. Also, CI seems not to be running because of conflicts with main.

File: src/codegen/operations.rs [hunk 2]
```diff
@@ -596,3 +598,58 @@ fn codegen_jumpdest<'c>(
...
+        flag,
+        &ok_block,
+        &op_ctx.revert_block,
+        &[],
+        &[],
+        location,
+    ));
+
+    let mut mask: [u8; 32] = [0; 32];
+   mask[31] = 0xff;
```
Comment (MegaRedHand) [inline src/codegen/operations.rs:2049]: The mask should be `[0xff; 32]` (the 256 bits set) for it to work.

File: src/codegen/operations.rs [hunk 3]
```diff
@@ -596,3 +598,58 @@ fn codegen_jumpdest<'c>(
...
+    let lhs = stack_pop(context, &ok_block)?;
+    // let one = ok_block
+    //     .append_operation(arith::constant(
+    //         context,
+    //         integer_constant_from_i64(context, 1).into(),
+    //         location,
+    //     ))
+    //     .result(0)?
+    //     .into();
```
Comment (MegaRedHand) [inline src/codegen/operations.rs:2058]: We can remove this

**Review by MegaRedHand — COMMENTED** [review]
Left you some comments. Once they're addressed I think it'll be ready to merge 😄
Please move the opcode from "not implemented" to "implemented" on the readme.

File: src/codegen/operations.rs (hunk 2 above)
Comment (mahmudsudo) [reply src/codegen/operations.rs:2049]: done

Comment (mahmudsudo) [reply src/codegen/operations.rs:2049]: setting the mask to [0xff;32] makes the test fail whilst 
`
 let mut mask: [u8; 32] = [0; 32];
   mask[31] = 0xff;
`
passes all tests

**Comment by mahmudsudo** [comment]
#216 reimplements this in line with the changes to the codebase

**Pull request closed without merging**